### PR TITLE
{bio}[foss/2018b] limix v2.0.4, bgen-reader v3.0.2 w/ Python v3.6.6 plus dependencies

### PR DESCRIPTION
**Pull request by lexming**
Wrapper PR for #7456

I have cleaned up quite thoroughly the PR, using as much dependencies from the tree as possible.

Completetly new easyconfigs:
* limix-2.0.4-foss-2018b-Python-3.6.6.eb 
* bgen-reader-3.0.2-foss-2018b-Python-3.6.6.eb
* liknorm-1.5.2-GCCcore-7.3.0.eb
* logaddexp-1.0.3-GCCcore-7.3.0.eb
* chi2comb-0.0.3-GCCcore-7.3.0.eb 

New easyconfigs with other versions already in tree:
* almosthere-1.0.1-GCCcore-7.3.0.eb 
* bgen-3.0.2-GCCcore-7.3.0.eb
* cytoolz-0.10.1-foss-2018b-Python-3.6.6.eb
* netcdf4-python-1.4.1-foss-2018b-Python-3.6.6.eb
* numba-0.39.0-foss-2018b-Python-3.6.6.eb
* statsmodels-0.9.0-foss-2018b-Python-3.6.6.eb
* xarray-0.12.1-foss-2018b-Python-3.6.6.eb
* zstd-1.4.0-GCCcore-7.3.0.eb 

Moreover, I also updated a couple of existing easyconfigs in the tree
1. rename the existing easyconfig `limix-bgen` to `bgen`. `limix` is the name of the developer, not the software and there are no other `bgen` packages that could cause confusion.
2. I'm also making some minor changes to almosthere-1.0.10-GCCcore-9.3.0.eb  to standarize all easyconfigs of `almosthere`.

### CODE REVIEW

File: easybuild/easyconfigs/a/almosthere/almosthere-1.0.1-GCCcore-7.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+source_urls = [GITHUB_SOURCE]
+sources = ['%(version)s.tar.gz']
+checksums = ['f9bc5d76370548dcdc5383063fd1896b35c5f64433166f5ea67708bfeb69c832']
+
+builddependencies = [
+    ('binutils', '2.30'),
+    ('CMake', '3.12.1'),
+]
+
+configopts = '-DBUILD_SHARED_LIBS=ON'
```
Comment (boegel) [inline easybuild/easyconfigs/a/almosthere/almosthere-1.0.1-GCCcore-7.3.0.eb:26]: Please also make this

```python
configopts = ['-DBUILD_SHARED_LIBS=ON', '-DBUILD_SHARED_LIBS=OFF']
```

File: easybuild/easyconfigs/x/xarray/xarray-0.12.1-foss-2018b-Python-3.6.6.eb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+checksums = ['ac09a819e791be208ae33fa7ecee19d0fe7b5479906f927d358a61957ce27e10']
+
+dependencies = [('Python', '3.6.6')]
+
+download_dep_fail = True
+use_pip = True
```
Comment (boegel) [inline easybuild/easyconfigs/x/xarray/xarray-0.12.1-foss-2018b-Python-3.6.6.eb:21]: Missing `sanity_pip_check = True`

File: easybuild/easyconfigs/s/statsmodels/statsmodels-0.9.0-foss-2018b-Python-3.6.6.eb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    ('patsy', '0.5.1', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/patsy'],
+        'checksums': ['f115cec4201e1465cd58b9866b0b0e7b941caafec129869057405bfe5b5e3991'],
+    }),
+    (name, version, {
+        'source_urls': ['https://pypi.python.org/packages/source/s/statsmodels'],
+        'checksums': ['6461f93a842c649922c2c9a9bc9d9c4834110b89de8c4af196a791ab8f42ba3b'],
+    }),
+]
+
```
Comment (boegel) [inline easybuild/easyconfigs/s/statsmodels/statsmodels-0.9.0-foss-2018b-Python-3.6.6.eb:28]: Missing `sanity_pip_check = True`

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/a/almosthere/almosthere-1.0.1-GCCcore-7.3.0.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/a/almosthere/almosthere-1.0.1-GCCcore-7.3.0.eb:26]: It's not necessary in this version, both the static and shared libs are created in one go with `-DBUILD_SHARED_LIBS=ON`.

File: easybuild/easyconfigs/x/xarray/xarray-0.12.1-foss-2018b-Python-3.6.6.eb (hunk 2 above)
Comment (lexming) [reply easybuild/easyconfigs/x/xarray/xarray-0.12.1-foss-2018b-Python-3.6.6.eb:21]: fixed :+1:

File: easybuild/easyconfigs/s/statsmodels/statsmodels-0.9.0-foss-2018b-Python-3.6.6.eb (hunk 3 above)
Comment (lexming) [reply easybuild/easyconfigs/s/statsmodels/statsmodels-0.9.0-foss-2018b-Python-3.6.6.eb:29]: fixed :+1:

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 16 out of 17 (16 easyconfigs in total)
node367.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/a4b6629ab4d8cad49c7d2d6b01e09042 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 16 out of 16 (16 easyconfigs in total)
node379.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/de1acb0550998f8d2e8028c747b47c34 for a full test report.

**Comment by zao** [comment]
Test report by @zao
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
eb-u1604 - Linux ubuntu 16.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.12
See https://gist.github.com/6644b3f081a253b28a08b02d2c91c4cc for a full test report.

**Comment by lexming** [comment]
Depends on https://github.com/easybuilders/easybuild-easyconfigs/pull/11873 , which fixes failed test in https://github.com/easybuilders/easybuild-easyconfigs/pull/11867#issuecomment-744040797

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 16 out of 16 (16 easyconfigs in total)
eb-u1604 - Linux ubuntu 16.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 2.7.12
See https://gist.github.com/c7ab5442a37484fa9270fe3fda97cdc5 for a full test report.

**Comment by zao** [comment]
The successful test report of mine above was made with the fixed LLVM.

**Review by boegel — APPROVED** [review]
lgtm, more test reports coming in...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 16 out of 16 (16 easyconfigs in total)
node3607.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/dd9618601f306c206d781b50432160de for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 16 out of 16 (16 easyconfigs in total)
node3106.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/70cd7b1957e3821abe53ab7151bd3edf for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 16 out of 16 (16 easyconfigs in total)
node2658.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/bfcbb3a5fdc0aff224a0bfa3b7a90b2f for a full test report.

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**